### DISCUSSION TITLE
Feature request? - Routing decision of interface weight

### DISCUSSION THREAD

**Discussion by SebastianObi** [markqvist/Reticulum#86] [Ideas]
Hello Mark,

I have the following question or feature request. I am not sure if something like this already exists or is planned. In the documentation I could not find anything in this direction.

The following scenario:
A node has 2 interfaces, 1x fast IP connection to the test network or backbone and 1x LoRa. At the moment the routing takes is of the basis of the lowest hop count. Which makes sense. But for this scenario it would be great to work alsi as the following: 
If the fast IP connection has 1 or more hops than the LoRa connection, the IP connection should still be preferred. This could be realized with a weight or metric setting in the interfaces. Then with received announce over an interface with higher weight this would be stored in the routing table. This would have a faster performance although the hop count is higher.

At this point you could also add the functionality that a default and fallback route is stored in the routing table. Then the next interface could be used directly if the primary interface fails or if down.

This should then be generally switchable on and off in the settings to adapt the routing behavior to the circumstances of the user.

I don't know if my idea is good or bad for a mesh routing protocol.

Have you ever thought about such functions?

**Comment by markqvist** [comment]
Just a quick note to say I read this :) I have actually been adding the foundation for supporting bw cost based metrics during the last few releases, so it is very close to being a "minor" feature to add. Let's discuss it in more detail when I am back online in two weeks!
